hooks/androidCopyPreferencesPermissions.js

```diff
@@ -0,0 +1,713 @@
+                addEntryToPermissionsXML(permissionsXmlDoc, arrayElement, permission.name)
+            });
+            
+            profilePermissionsWrite.forEach(permission => {
+                addEntryToManifest(manifestXmlDoc, permission.name)
+                addEntryToPermissionsXML(permissionsXmlDoc, arrayElement, permission.name)
+            });
+
+        }
+
+    // Serialize the updated XML document back to string
+    const serializer = new XMLSerializer();
+
+    // Android >= 14
+    const updatedManifestXmlString = serializer.serializeToString(manifestXmlDoc);
+    // Write the updated XML string back to the same file
+    fs.writeFileSync(manifestFilePath, updatedManifestXmlString, 'utf-8');
+
+    // Android <= 13
+    const updatedPermissionsXmlString = serializer.serializeToString(permissionsXmlDoc);
+    const permissionsXmlFilePath = path.join(projectRoot, 'platforms/android/app/src/main/res/values/health_permissions.xml');
+    // Write the updated XML string back to the same file
+    fs.writeFileSync(permissionsXmlFilePath, updatedPermissionsXmlString, 'utf-8');
+
+}
+
+function addBackgroundJobPermissionsToManifest(configParser, projectRoot, parser) {
+
+    const disableBackgroundJobs = configParser.getPlatformPreference('DisableBackgroundJobs', 'android');
+
+    // we want to include the permissions by default
+    // if disableBackgroundJobs == true then we don't want to include the permissions in the manfiest
+    if (disableBackgroundJobs !== "true") {
+
+        const manifestFilePath = path.join(projectRoot, 'platforms/android/app/src/main/AndroidManifest.xml');
+        const manifestXmlString = fs.readFileSync(manifestFilePath, 'utf-8');
+
+        // Parse the XML string
+        const manifestXmlDoc = parser.parseFromString(manifestXmlString, 'text/xml');
+
+        // add permissions to XML document
+        addEntryToManifest(manifestXmlDoc, 'android.permission.POST_NOTIFICATIONS')
+        addEntryToManifest(manifestXmlDoc, 'android.permission.ACTIVITY_RECOGNITION')
+        addEntryToManifest(manifestXmlDoc, 'android.permission.FOREGROUND_SERVICE')
+        addEntryToManifest(manifestXmlDoc, 'android.permission.FOREGROUND_SERVICE_HEALTH')
+        addEntryToManifest(manifestXmlDoc, 'android.permission.HIGH_SAMPLING_RATE_SENSORS')
+
+        // serialize the updated XML document back to string
+        const serializer = new XMLSerializer();
+        const updatedManifestXmlString = serializer.serializeToString(manifestXmlDoc);
+
+        // write the updated XML string back to the same file
+        fs.writeFileSync(manifestFilePath, updatedManifestXmlString, 'utf-8');
+    }
+
+}
+
+function addEntryToManifest(manifestXmlDoc, permission) {
+    const newPermission = manifestXmlDoc.createElement('uses-permission');
+    newPermission.setAttribute('android:name', permission);
+    manifestXmlDoc.documentElement.appendChild(newPermission);
+}
```

Comment:
If this type of util function is going to be useful across multiple repos/plugins maybe we pull these functions out into a npm typed lib package?

Reply:
For sure, if we have to use this kind of function again I think we should do that!